frontend/cypress/integration/common/workload_logs.ts

```diff
@@ -0,0 +1,25 @@
+import { Given, Then, When } from "cypress-cucumber-preprocessor/steps";
+
+Given("I am on the {string} workload detail page", (workload) => {
+    // Go to workloads list
+    cy.get("#Workloads").click();
+
+    // Select "bookinfo" namespace
+    cy.get("#namespace-selector").click();
+    cy.get("#bulk-select-id").uncheck();
+    cy.get('#namespace-list-item\\[bookinfo\\] input').check();
+    cy.get("#namespace-selector").click();
+
+    // Open workload detail page
+    cy.get(`[data-cy=VirtualItem_Nsbookinfo_${workload}] a`).click();
+});
+
+When("I go to the Logs tab of the workload detail page", () => {
```

Comment:
Same here, if possible, passing the namespace and workload name as parameters may help to reuse this expression.

Reply:
This is supposed to be used when the view is already the detail page. So, the test is supposed to have the form:
```
Given I am on the "productpage-v1" workload detail page of the "bookinfo" namespace
When I go to the Logs tab of the workload detail page
Then ...
```

Adding the namespace and the workload would be redundant and would be unused under the assumption that the view is already a detail page...

So, I think this can be easily reused as long as the proper `Given` expressions are used.